src/plugins/regions.ts

```diff
@@ -313,22 +313,29 @@ class RegionsPlugin extends BasePlugin<RegionsPluginEvents, RegionsPluginOptions
     }
     this.wavesurfer.getWrapper().appendChild(this.regionsContainer)
 
-    // Detect when a region is being played
-    let activeRegion: Region | null = null
-
+    let activeRegions: Region[] = []
     this.subscriptions.push(
       this.wavesurfer.on('timeupdate', (currentTime) => {
-        const playedRegion = this.regions.find((region) => region.start <= currentTime && region.end >= currentTime)
-
-        if (activeRegion && activeRegion !== playedRegion) {
-          this.emit('region-out', activeRegion)
-          activeRegion = null
-        }
-
-        if (playedRegion && playedRegion !== activeRegion) {
-          activeRegion = playedRegion
-          this.emit('region-in', playedRegion)
-        }
+        // Detect when regions are being played
+        const playedRegions = this.regions.filter((region) => region.start <= currentTime && region.end >= currentTime)
+
+        // Trigger region-in when activeRegions doesn't include a played regions
+        playedRegions.forEach((region) => {
+          if (!activeRegions.includes(region)) {
+            activeRegions = [...activeRegions, region]
```

Comment:
Here, too, you can just assign `activeRegions = playedRegions`.

Reply:
Oh, it's my mistake!!
that are same